diff --git a/src/dynamic_frame_tf2_broadcaster.cpp b/src/dynamic_frame_tf2_broadcaster.cpp
index de08ba7..7e5a7b7 100644
--- a/src/dynamic_frame_tf2_broadcaster.cpp
+++ b/src/dynamic_frame_tf2_broadcaster.cpp
@@ -30,7 +30,7 @@ public:
   DynamicFrameBroadcaster()
   : Node("dynamic_frame_tf2_broadcaster")
   {
-    tf_broadcaster_ = std::make_shared<tf2_ros::TransformBroadcaster>(this);
+    tf_broadcaster_ = std::make_shared<tf2_ros::TransformBroadcaster>(*this);
     timer_ = this->create_wall_timer(
       100ms, std::bind(&DynamicFrameBroadcaster::broadcast_timer_callback, this));
   }
diff --git a/src/fixed_frame_tf2_broadcaster.cpp b/src/fixed_frame_tf2_broadcaster.cpp
index 057134f..47347e8 100644
--- a/src/fixed_frame_tf2_broadcaster.cpp
+++ b/src/fixed_frame_tf2_broadcaster.cpp
@@ -28,7 +28,7 @@ public:
   FixedFrameBroadcaster()
   : Node("fixed_frame_tf2_broadcaster")
   {
-    tf_broadcaster_ = std::make_shared<tf2_ros::TransformBroadcaster>(this);
+    tf_broadcaster_ = std::make_shared<tf2_ros::TransformBroadcaster>(*this);
     timer_ = this->create_wall_timer(
       100ms, std::bind(&FixedFrameBroadcaster::broadcast_timer_callback, this));
   }
diff --git a/src/static_turtle_tf2_broadcaster.cpp b/src/static_turtle_tf2_broadcaster.cpp
index ddd3218..8f25849 100644
--- a/src/static_turtle_tf2_broadcaster.cpp
+++ b/src/static_turtle_tf2_broadcaster.cpp
@@ -25,7 +25,7 @@ public:
   explicit StaticFramePublisher(char * transformation[])
   : Node("static_turtle_tf2_broadcaster")
   {
-    tf_static_broadcaster_ = std::make_shared<tf2_ros::StaticTransformBroadcaster>(this);
+    tf_static_broadcaster_ = std::make_shared<tf2_ros::StaticTransformBroadcaster>(*this);
 
     // Publish static transforms once at startup
     this->make_transforms(transformation);
diff --git a/src/turtle_tf2_message_filter.cpp b/src/turtle_tf2_message_filter.cpp
index c2d7085..f57ff6f 100644
--- a/src/turtle_tf2_message_filter.cpp
+++ b/src/turtle_tf2_message_filter.cpp
@@ -45,17 +45,14 @@ public:
     tf2_buffer_ = std::make_shared<tf2_ros::Buffer>(this->get_clock());
     // Create the timer interface before call to waitForTransform,
     // to avoid a tf2_ros::CreateTimerInterfaceException exception
-    auto timer_interface = std::make_shared<tf2_ros::CreateTimerROS>(
-      this->get_node_base_interface(),
-      this->get_node_timers_interface());
+    auto timer_interface = std::make_shared<tf2_ros::CreateTimerROS>(*this);
     tf2_buffer_->setCreateTimerInterface(timer_interface);
     tf2_listener_ =
       std::make_shared<tf2_ros::TransformListener>(*tf2_buffer_);
 
     point_sub_.subscribe(this, "/turtle3/turtle_point_stamped", rclcpp::QoS(10));
     tf2_filter_ = std::make_shared<tf2_ros::MessageFilter<geometry_msgs::msg::PointStamped>>(
-      point_sub_, *tf2_buffer_, target_frame_, 100, this->get_node_logging_interface(),
-      this->get_node_clock_interface(), buffer_timeout);
+      point_sub_, *tf2_buffer_, target_frame_, 100, *this, buffer_timeout);
     // Register a callback with tf2_ros::MessageFilter to be called when transforms are available
     tf2_filter_->registerCallback(&PoseDrawer::msgCallback, this);
   }
