diff --git a/src/FloorplanVisualizer.cpp b/src/FloorplanVisualizer.cpp
index 15c07c0..6c6035e 100644
--- a/src/FloorplanVisualizer.cpp
+++ b/src/FloorplanVisualizer.cpp
@@ -74,7 +74,9 @@ FloorplanVisualizer::FloorplanVisualizer(const rclcpp::NodeOptions& options)
           continue;
 
         cv::Mat cv_img = cv::imdecode(
-          cv::Mat(level.images[0].data), cv::IMREAD_GRAYSCALE);
+          cv::Mat(std::vector<unsigned char>(
+            level.images[0].data.begin(), level.images[0].data.end())),
+          cv::IMREAD_GRAYSCALE);
         auto it = level.images.begin();
         ++it;
         // We blend all the other images into the first image
@@ -84,7 +86,9 @@ FloorplanVisualizer::FloorplanVisualizer(const rclcpp::NodeOptions& options)
         for (; it != level.images.end(); ++it)
         {
           cv::Mat next_img = cv::imdecode(
-            cv::Mat(it->data), cv::IMREAD_GRAYSCALE);
+            cv::Mat(std::vector<unsigned char>(
+              it->data.begin(), it->data.end())),
+            cv::IMREAD_GRAYSCALE);
           cv::addWeighted(cv_img, 0.7, next_img, 0.3, 0.0, cv_img);
         }
         const auto& image = level.images[0];
