diff --git a/src/controller_manager_plugin.cpp b/src/controller_manager_plugin.cpp
index 4f24024725..3fc2cee2e0 100644
--- a/src/controller_manager_plugin.cpp
+++ b/src/controller_manager_plugin.cpp
@@ -54,6 +54,7 @@
 #include <map>
 #include <memory>
 #include <queue>
+#include <unordered_set>
 #include <moveit/utils/logger.hpp>
 
 static const rclcpp::Duration CONTROLLER_INFORMATION_VALIDITY_AGE = rclcpp::Duration::from_seconds(1.0);
