diff --git a/src/rosidlcpp_generator_py/rosidlcpp_generator_py.cpp b/src/rosidlcpp_generator_py/rosidlcpp_generator_py.cpp
--- a/src/rosidlcpp_generator_py/rosidlcpp_generator_py.cpp
+++ b/src/rosidlcpp_generator_py/rosidlcpp_generator_py.cpp
@@ -22,6 +22,7 @@
 
 #include <fmt/core.h>
 #include <fmt/format.h>
+#include <fmt/ranges.h>
 
 #include <nlohmann/json.hpp>
 #include <nlohmann/json_fwd.hpp>
