diff --git a/CMakeLists.txt b/CMakeLists.txt
index b9ca55c..c88adac 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -37,13 +37,15 @@ ament_auto_add_library(${PROJECT_NAME} SHARED
   src/udp_socket.cpp
   src/udp_driver.cpp
 )
-ament_target_dependencies(${PROJECT_NAME} "ASIO")
+target_include_directories(${PROJECT_NAME} PUBLIC ${ASIO_INCLUDE_DIRS})
+target_compile_definitions(${PROJECT_NAME} PUBLIC ${ASIO_DEFINITIONS})
 
 ament_auto_add_library(${PROJECT_NAME}_nodes SHARED
   src/udp_receiver_node.cpp
   src/udp_sender_node.cpp
 )
-ament_target_dependencies(${PROJECT_NAME}_nodes "ASIO")
+target_include_directories(${PROJECT_NAME}_nodes PUBLIC ${ASIO_INCLUDE_DIRS})
+target_compile_definitions(${PROJECT_NAME}_nodes PUBLIC ${ASIO_DEFINITIONS})
 target_link_libraries(${PROJECT_NAME}_nodes ${PROJECT_NAME})
 
 rclcpp_components_register_node(${PROJECT_NAME}_nodes
@@ -61,7 +63,8 @@ ament_auto_add_executable(udp_bridge_node_exe
 )
 
 target_link_libraries(udp_bridge_node_exe ${PROJECT_NAME} ${PROJECT_NAME}_nodes)
-ament_target_dependencies(udp_bridge_node_exe ASIO)
+target_include_directories(udp_bridge_node_exe PUBLIC ${ASIO_INCLUDE_DIRS})
+target_compile_definitions(udp_bridge_node_exe PUBLIC ${ASIO_DEFINITIONS})
 
 if(BUILD_TESTING)
   find_package(ament_lint_auto REQUIRED)
diff --git a/src/udp_socket.cpp b/src/udp_socket.cpp
index 619c7be..1b12400 100644
--- a/src/udp_socket.cpp
+++ b/src/udp_socket.cpp
@@ -38,15 +38,15 @@ UdpSocket::UdpSocket(
   const uint16_t host_port)
 : m_ctx(ctx),
   m_udp_socket(ctx.ios()),
-  m_remote_endpoint(address::from_string(remote_ip), remote_port),
-  m_host_endpoint(address::from_string(host_ip), host_port)
+  m_remote_endpoint(asio::ip::make_address(remote_ip), remote_port),
+  m_host_endpoint(asio::ip::make_address(host_ip), host_port)
 {
   m_remote_endpoint = remote_ip.empty() ?
     udp::endpoint{udp::v4(), remote_port} :
-  udp::endpoint{address::from_string(remote_ip), remote_port};
+  udp::endpoint{asio::ip::make_address(remote_ip), remote_port};
   m_host_endpoint = host_ip.empty() ?
     udp::endpoint{udp::v4(), host_port} :
-  udp::endpoint{address::from_string(host_ip), host_port};
+  udp::endpoint{asio::ip::make_address(host_ip), host_port};
   m_recv_buffer.resize(m_recv_buffer_size);
 }
 
