diff --git a/include/parameter_traits/parameter_traits.hpp b/include/parameter_traits/parameter_traits.hpp
index a62153dc..f190922b 100644
--- a/include/parameter_traits/parameter_traits.hpp
+++ b/include/parameter_traits/parameter_traits.hpp
@@ -34,7 +34,7 @@
 #include <vector>
 
 #include <rsl/algorithm.hpp>
-#include <tl_expected/expected.hpp>
+#include <tl/expected.hpp>
 
 #ifndef SILENCE_DEPRECATION_WARNINGS
 #ifdef _MSC_VER
