diff --git a/deps/libmotioncapture/deps/NatNetSDKCrossplatform/CMakeLists.txt b/deps/libmotioncapture/deps/NatNetSDKCrossplatform/CMakeLists.txt
index 5dc394f..d577449 100644
--- a/deps/libmotioncapture/deps/NatNetSDKCrossplatform/CMakeLists.txt
+++ b/deps/libmotioncapture/deps/NatNetSDKCrossplatform/CMakeLists.txt
@@ -5,7 +5,7 @@ set(CMAKE_THREAD_PREFER_PTHREAD TRUE)
 set(THREADS_PREFER_PTHREAD_FLAG TRUE)
 find_package(Threads REQUIRED)
 
-find_package(Boost 1.5 REQUIRED COMPONENTS system thread)
+find_package(Boost 1.5 REQUIRED COMPONENTS thread)
 
 # Enable C++14 and warnings
 set(CMAKE_CXX_STANDARD 14)
@@ -21,7 +21,6 @@ add_executable(packetClient
   samples/PacketClient/PacketClient.cpp
 )
 target_link_libraries(packetClient
-  Boost::system
   Boost::thread
 )
 
diff --git a/deps/libmotioncapture/deps/pybind11/CMakeLists.txt b/deps/libmotioncapture/deps/pybind11/CMakeLists.txt
index 0d93203..5b59677 100644
--- a/deps/libmotioncapture/deps/pybind11/CMakeLists.txt
+++ b/deps/libmotioncapture/deps/pybind11/CMakeLists.txt
@@ -5,7 +5,7 @@
 # All rights reserved. Use of this source code is governed by a
 # BSD-style license that can be found in the LICENSE file.
 
-cmake_minimum_required(VERSION 3.4)
+cmake_minimum_required(VERSION 3.5)
 
 # The `cmake_minimum_required(VERSION 3.4...3.22)` syntax does not work with
 # some versions of VS that have a patched CMake 3.11. This forces us to emulate
diff --git a/deps/libmotioncapture/deps/pybind11/tests/CMakeLists.txt b/deps/libmotioncapture/deps/pybind11/tests/CMakeLists.txt
index 491f215..f0a9112 100644
--- a/deps/libmotioncapture/deps/pybind11/tests/CMakeLists.txt
+++ b/deps/libmotioncapture/deps/pybind11/tests/CMakeLists.txt
@@ -5,7 +5,7 @@
 # All rights reserved. Use of this source code is governed by a
 # BSD-style license that can be found in the LICENSE file.
 
-cmake_minimum_required(VERSION 3.4)
+cmake_minimum_required(VERSION 3.5)
 
 # The `cmake_minimum_required(VERSION 3.4...3.18)` syntax does not work with
 # some versions of VS that have a patched CMake 3.11. This forces us to emulate
diff --git a/deps/libmotioncapture/deps/pybind11/tests/test_cmake_build/installed_embed/CMakeLists.txt b/deps/libmotioncapture/deps/pybind11/tests/test_cmake_build/installed_embed/CMakeLists.txt
index f7d6939..2847142 100644
--- a/deps/libmotioncapture/deps/pybind11/tests/test_cmake_build/installed_embed/CMakeLists.txt
+++ b/deps/libmotioncapture/deps/pybind11/tests/test_cmake_build/installed_embed/CMakeLists.txt
@@ -1,4 +1,4 @@
-cmake_minimum_required(VERSION 3.4)
+cmake_minimum_required(VERSION 3.5)
 
 # The `cmake_minimum_required(VERSION 3.4...3.18)` syntax does not work with
 # some versions of VS that have a patched CMake 3.11. This forces us to emulate
diff --git a/deps/libmotioncapture/deps/pybind11/tests/test_cmake_build/installed_function/CMakeLists.txt b/deps/libmotioncapture/deps/pybind11/tests/test_cmake_build/installed_function/CMakeLists.txt
index d7ca4db..74322f5 100644
--- a/deps/libmotioncapture/deps/pybind11/tests/test_cmake_build/installed_function/CMakeLists.txt
+++ b/deps/libmotioncapture/deps/pybind11/tests/test_cmake_build/installed_function/CMakeLists.txt
@@ -1,4 +1,4 @@
-cmake_minimum_required(VERSION 3.4)
+cmake_minimum_required(VERSION 3.5)
 project(test_installed_module CXX)
 
 # The `cmake_minimum_required(VERSION 3.4...3.18)` syntax does not work with
diff --git a/deps/libmotioncapture/deps/pybind11/tests/test_cmake_build/installed_target/CMakeLists.txt b/deps/libmotioncapture/deps/pybind11/tests/test_cmake_build/installed_target/CMakeLists.txt
index bc5e101..2d21a21 100644
--- a/deps/libmotioncapture/deps/pybind11/tests/test_cmake_build/installed_target/CMakeLists.txt
+++ b/deps/libmotioncapture/deps/pybind11/tests/test_cmake_build/installed_target/CMakeLists.txt
@@ -1,4 +1,4 @@
-cmake_minimum_required(VERSION 3.4)
+cmake_minimum_required(VERSION 3.5)
 
 # The `cmake_minimum_required(VERSION 3.4...3.18)` syntax does not work with
 # some versions of VS that have a patched CMake 3.11. This forces us to emulate
diff --git a/deps/libmotioncapture/deps/pybind11/tests/test_cmake_build/subdirectory_embed/CMakeLists.txt b/deps/libmotioncapture/deps/pybind11/tests/test_cmake_build/subdirectory_embed/CMakeLists.txt
index 58cdd7c..f9835f3 100644
--- a/deps/libmotioncapture/deps/pybind11/tests/test_cmake_build/subdirectory_embed/CMakeLists.txt
+++ b/deps/libmotioncapture/deps/pybind11/tests/test_cmake_build/subdirectory_embed/CMakeLists.txt
@@ -1,4 +1,4 @@
-cmake_minimum_required(VERSION 3.4)
+cmake_minimum_required(VERSION 3.5)
 
 # The `cmake_minimum_required(VERSION 3.4...3.18)` syntax does not work with
 # some versions of VS that have a patched CMake 3.11. This forces us to emulate
diff --git a/deps/libmotioncapture/deps/pybind11/tests/test_cmake_build/subdirectory_function/CMakeLists.txt b/deps/libmotioncapture/deps/pybind11/tests/test_cmake_build/subdirectory_function/CMakeLists.txt
index 01557c4..6d96cc3 100644
--- a/deps/libmotioncapture/deps/pybind11/tests/test_cmake_build/subdirectory_function/CMakeLists.txt
+++ b/deps/libmotioncapture/deps/pybind11/tests/test_cmake_build/subdirectory_function/CMakeLists.txt
@@ -1,4 +1,4 @@
-cmake_minimum_required(VERSION 3.4)
+cmake_minimum_required(VERSION 3.5)
 
 # The `cmake_minimum_required(VERSION 3.4...3.18)` syntax does not work with
 # some versions of VS that have a patched CMake 3.11. This forces us to emulate
diff --git a/deps/libmotioncapture/deps/pybind11/tests/test_cmake_build/subdirectory_target/CMakeLists.txt b/deps/libmotioncapture/deps/pybind11/tests/test_cmake_build/subdirectory_target/CMakeLists.txt
index ba82fde..6f8e044 100644
--- a/deps/libmotioncapture/deps/pybind11/tests/test_cmake_build/subdirectory_target/CMakeLists.txt
+++ b/deps/libmotioncapture/deps/pybind11/tests/test_cmake_build/subdirectory_target/CMakeLists.txt
@@ -1,4 +1,4 @@
-cmake_minimum_required(VERSION 3.4)
+cmake_minimum_required(VERSION 3.5)
 
 # The `cmake_minimum_required(VERSION 3.4...3.18)` syntax does not work with
 # some versions of VS that have a patched CMake 3.11. This forces us to emulate
