diff --git a/src/voxel_grid_downsample_filter/voxel_grid_downsample_filter_node.cpp b/src/voxel_grid_downsample_filter/voxel_grid_downsample_filter_node.cpp
index 241239d..4b02608 100644
--- a/src/voxel_grid_downsample_filter/voxel_grid_downsample_filter_node.cpp
+++ b/src/voxel_grid_downsample_filter/voxel_grid_downsample_filter_node.cpp
@@ -21,7 +21,6 @@
 #include <pcl_ros/transforms.hpp>
 #include <tf2_eigen/tf2_eigen.hpp>
 
-#include <pcl/io/io.h>
 #include <pcl/kdtree/kdtree_flann.h>
 #include <pcl/search/kdtree.h>
 #include <pcl/segmentation/segment_differences.h>
